File: 724-find-pivot-index/724-find-pivot-index.py
from itertools import accumulate
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        nums[::] = accumulate(nums)
        total = nums[-1]
        
        prev = 0
        for (i, n) in enumerate(nums):
            if i == 0:
                if total- n == 0:
                    return i
            elif i == len(nums) - 1:
                if prev == 0:
                    return i
            else:
                if prev == total - n:
                    return i
                
            prev = n
            
        return -1
                    
            

File: 724-find-pivot-index/724-find-pivot-index.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        
        let mut l = 0;
        let mut r: i32 = nums.iter().sum();
        
        for (i, n) in nums.into_iter().enumerate(){
            r -= n;
            
            if l == r{
                return i as i32;
            }
            
            l += n;
        }
        
        -1
    }
}